crates/subspace-core-primitives/src/lib.rs

```diff
@@ -215,6 +222,144 @@ impl PosProof {
     pub const SIZE: usize = 17 * 8;
 }
 
+/// Proof of time key(input to the encryption).
+#[derive(Debug, Copy, Clone, Encode, Decode)]
```

Comment:
I think all of these data structures will need to derive `TypeInfo`, `MaxEncodedLen` and serde traits too (as hex, see other examples of byte containers).

Reply:
It is just two derives away that are already imported in this module, why waiting :shrug:

Reply:
this falls in same bucket as the consensus changes that were reverted, let us do these together